UncomplicatedCustomItems/API/Features/SummonedCustomItem.cs

```diff
@@ -0,0 +1,208 @@
+            Serial = Item.Serial;
+            IsPickup = false;
+            Pickup = null;
+        }
+
+        /// <summary>
+        /// Summon the <see cref="ICustomItem"/> as an existing pickup
+        /// </summary>
+        /// <param name="customItem"></param>
+        /// <param name="pickup"></param>
+        public SummonedCustomItem(ICustomItem customItem, Pickup pickup)
+        {
+            CustomItem = customItem;
+            Owner = null;
+            Pickup = pickup;
+            Serial = Pickup.Serial;
+            IsPickup = true;
+            Item = null;
+        }
+
+        /// <summary>
+        /// Summon the <see cref="ICustomItem"/> as a new pickup
+        /// </summary>
+        /// <param name="customItem"></param>
+        /// <param name="position"></param>
+        /// <param name="rotation"></param>
+        public SummonedCustomItem(ICustomItem customItem, Vector3 position, Quaternion rotation = new())
+        {
+            CustomItem = customItem;
+            Owner = null;
+            Item = null;
+            Pickup = Pickup.CreateAndSpawn(customItem.Item, position, rotation);
+            IsPickup = true;
+            Serial = Pickup.Serial;
+        }
+
+        /// <summary>
+        /// Summon the <see cref="ICustomItem"/> inside the inventory of a player
+        /// </summary>
+        /// <param name="customItem"></param>
+        /// <param name="owner"></param>
+        /// <returns>The <see cref="SummonedCustomItem"/> class of the summoned item</returns>
+        public static SummonedCustomItem Summon(ICustomItem customItem, Player owner)
+        {
+            return new(customItem, owner);
+        }
+
+        internal void SetProperties()
+        {
+            if (Item is not null)
+            {
+                Item.Scale = CustomItem.Scale;
+                switch (CustomItem.CustomItemType)
+                {
+                    case CustomItemType.Item:
+                        break;
+
+                    case CustomItemType.Keycard:
+                        ((Keycard)Item).Permissions = ((IKeycardData)CustomItem.CustomData).Permissions;
+                        break;
+
+                    case CustomItemType.Armor:
+                        ((Armor)Item).HelmetEfficacy = ((IArmorData)CustomItem.CustomData).HeadProtection;
+                        ((Armor)Item).RemoveExcessOnDrop = ((IArmorData)CustomItem.CustomData).RemoveExcessOnDrop;
+                        ((Armor)Item).StaminaUseMultiplier = ((IArmorData)CustomItem.CustomData).StaminaUseMultiplier;
+                        break;
+
+                    case CustomItemType.Weapon:
+                        ((Firearm)Item).MaxAmmo = ((IWeaponData)CustomItem.CustomData).MaxAmmo;
+                        ((Firearm)Item).FireRate = ((IWeaponData)CustomItem.CustomData).FireRate;
+                        break;
+
+                    default:
+                        break;
+                }
+            }
```

Comment:
...

Reply:
Bro does not like casting

Reply:
I love casts if they are appropriate. But this is fucked up.

Reply:
Cast can be done in different ways.
This is less elegant but way more fast than the if cast

Reply:
Bro, these casts look disgusting

Reply:
But they're ok, an if would be inefficient and useless as we already know the type.
https://learn.microsoft.com/it-it/dotnet/csharp/programming-guide/types/casting-and-type-conversions

Reply:
Before that, I wrote a system where castes were not needed.

Reply:
But was unefficient, custom items were managed as different ones.
And casting is safe

Reply:
Why unefficient?

Reply:
and check dsicord

Reply:
> Why unefficient?

Because in the config were divided and the management was way more complex than this